storey/flow.py

```diff
@@ -1435,3 +1437,185 @@ def get_table(self, key):
+        :param path: Event path
+        """
+        return body
+
+    def _run(self, body: Any, path: str) -> Any:
+        start = time.monotonic()
+        body = self.run(body, path)
+        end = time.monotonic()
+        return _ParallelExecutionRunnableResult(self.name, body, end - start)
+
+    async def _async_run(self, body: Any, path: str) -> Any:
+        start = time.monotonic()
+        body = await self.run_async(body, path)
+        end = time.monotonic()
+        return _ParallelExecutionRunnableResult(self.name, body, end - start)
+
+
+class ParallelExecution(Flow):
+    """
+    Runs multiple jobs in parallel for each event.
+
+    :param runnables: A list of ParallelExecutionRunnable instances.
+    :param max_processes: Maximum number of processes to spawn. Defaults to the number of available CPUs, or 16 if
+      number of CPUs can't be determined.
+    :param max_threads: Maximum number of threads to start. Defaults to 32.
+    """
+
+    def __init__(
+        self,
+        runnables: list[ParallelExecutionRunnable],
+        max_processes: Optional[int] = None,
+        max_threads: Optional[int] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        if not runnables:
+            raise ValueError("ParallelExecution cannot be instantiated without at least one runnable")
+
+        self.runnables = runnables
+        self._runnable_by_name = {}
+
+        self.max_processes = max_processes or os.cpu_count() or 16
+        self.max_threads = max_threads or 32
+
+    def select_runnables(self, event) -> Optional[Union[list[str], list[ParallelExecutionRunnable]]]:
+        """
+        Given an event, returns a list of runnables (or a list of runnable names) to execute on it. It can also return
+        None, in which case all runnables are executed on the event, which is also the default.
+
+        :param event: Event object
+        """
+        pass
+
+    def _init(self):
+        super()._init()
+        num_processes = 0
+        num_threads = 0
+        for runnable in self.runnables:
+            if runnable.name in self._runnable_by_name:
+                raise ValueError(f"ParallelExecutionRunnable name '{runnable.name}' is not unique")
+            self._runnable_by_name[runnable.name] = runnable
+            print(f"initializing runnable {runnable.name}")
+            runnable.init()
+            if runnable.execution_mechanism == "multiprocessing":
+                num_processes += 1
+            elif runnable.execution_mechanism == "threading":
+                num_threads += 1
+            elif runnable.execution_mechanism not in ("asyncio", "naive"):
+                raise ValueError(f"Unsupported execution mechanism: {runnable.execution_mechanism}")
+
+        # enforce max
+        if self.max_processes:
+            num_processes = min(num_processes, self.max_processes)
+        if self.max_threads:
+            num_threads = min(num_threads, self.max_threads)
```

Comment:
The if statement can be remove - due to the new defaults.  
```suggestion
        num_processes = min(num_processes, self.max_processes)
        num_threads = min(num_threads, self.max_threads)
```